portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -1,143 +1,155 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
-import com.google.appengine.api.datastore.FetchOptions; 
-import com.google.appengine.api.datastore.FetchOptions.Builder; 
-import com.google.gson.Gson;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+import com.google.appengine.api.datastore.DatastoreServiceConfig.Builder;  
+import com.google.appengine.api.datastore.ReadPolicy;
+import com.google.appengine.api.datastore.ReadPolicy.Consistency;
 import com.google.sps.data.Quote; 
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
 import java.util.List; 
 import java.util.ArrayList; 
+import java.util.HashMap; 
 
-/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   private static final String QUOTE = "Quote";
   private static final String TEXT = "text";
   private static final String TIMESTAMP = "timestamp"; 
+  private static final String USEREMAIL = "userEmail";
+  private static final String LOGGEDIN = "loggedIn"; 
+  private static final String REDIRECTURL = "redirectUrl";
+  private static final String NICKNAME = "nickname";
   private static final int DEFAULT_NUM_QUOTES = 5;
 
-  /**
-    * Retrives all quotes that are stored in Datastore
-    * and puts them into a list of quotes. This list is then written
-    * as the response from the server. 
-    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Query query = new Query(QUOTE).addSort(TIMESTAMP, SortDirection.DESCENDING);
-    PreparedQuery quotes = datastore.prepare(query);
+    response.setContentType("application/json");
 
-    // Get the parameter for number to display, with 5 as the default value.
-    int numQuoteToDisplay = getNumQuoteDisplayed(request, DEFAULT_NUM_QUOTES);
-   
-    // Add each quote to the quote list whose content will be written as 
-    // the response from the servlet. 
-    List<Quote> quoteList = new ArrayList<>();
-    quotes.asList(FetchOptions.Builder.withLimit(numQuoteToDisplay))
-        .forEach((quoteEntity) -> {
-          quoteList.add(extractQuoteFromEntity(quoteEntity));
-        });
-    
-    // Put the thread to sleep for a short period of time.
-    // This adds buffer time for querying Datastore. 
-    try {
-      Thread.sleep(200);
-    } catch (Exception e) {
-      System.out.println(e);
+    UserService userService = UserServiceFactory.getUserService();
+    HashMap<String, String> responseMap = new HashMap<String, String>();
```

Comment:
We can use the interface as the type.

Map<...> ...

Reply:
Done.